lib/Serialization/SerializeDoc.cpp

```diff
@@ -618,6 +610,20 @@ class StringWriter {
   }
 };
 
+static void writePosition(const SourcePosition &Pos, endian::Writer &Writer,
+                          StringWriter &Strings) {
+  Writer.write<uint32_t>(Pos.Offset);
+  Writer.write<uint32_t>(Pos.Line);
+  Writer.write<uint32_t>(Pos.Column);
+
+  Writer.write<uint32_t>(Pos.Directive.Offset);
```

Comment:
I *could* make another table, but no, that's not quite what I was thinking.

The position only needs to be fixed size at the moment because the values stored in the on-disk hash table for decl records are indices. That doesn't need to be the case though, ie. we could store the offset into the blob and just have a marker for whether there is a directive or not.  This could just be length, if 0 we could skip reading offset/line offset/name.

I avoided doing this so as to minimize the serialization changes. Do you think this/your idea is worth doing now?

Reply:
I see.  It's hard to say without trying different approaches and benchmarking them.  Are you evaluating performance of these changes in some way?